sqlmesh/core/state_sync/engine_adapter.py

```diff
@@ -899,9 +899,6 @@ def _migrate_rows(self) -> None:
             if new_snapshot.fingerprint == snapshot.fingerprint:
                 logger.debug(f"{new_snapshot.snapshot_id} is unchanged.")
                 continue
-            if new_snapshot.snapshot_id in all_snapshots:
```

Comment:
Even if the snapshot already exists and there was a hash collission we should still preserver it in the mapping in order to update environments correctly.